diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index 87cd458..704d89f 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -37,6 +37,7 @@ EXPORT_TO_SOONG := \
 # Documentation here:
 # https://github.com/LineageOS/android_build_soong/commit/8328367c44085b948c003116c0ed74a047237a69
 
+include bootable/recovery/orangefox_soong.mk
 SOONG_CONFIG_NAMESPACES += twrpVarsPlugin
 
 SOONG_CONFIG_twrpVarsPlugin :=
