diff --git a/etc/init.rc b/etc/init.rc
index 5c13ef74..1333f745 100644
--- a/etc/init.rc
+++ b/etc/init.rc
@@ -171,6 +171,7 @@ on property:sys.usb.config=adb && property:sys.usb.configfs=0
     setprop sys.usb.state ${sys.usb.config}
 
 on property:sys.usb.config=sideload && property:sys.usb.configfs=0
+    write /sys/class/android_usb/android0/enable 0
     write /sys/class/android_usb/android0/idProduct D001
     write /sys/class/android_usb/android0/functions adb
     write /sys/class/android_usb/android0/enable 1
